core/src/common/empty_block/inherents/providers.rs

```diff
@@ -0,0 +1,167 @@
+        ))
+    }
+}
+
+impl<B: BlockT> InherentProvider<B> for ProviderVariant {
+    type Err = String;
+
+    fn get_inherent_providers_and_pre_digest(
+        &self,
+        maybe_parent_info: Option<(InherentData, Digest)>,
+        parent_header: B::Header,
+        ext: &mut TestExternalities<HashingFor<B>>,
+    ) -> InherentProviderResult<Self::Err> {
+        match *self {
+            ProviderVariant::Smart => {
+                 <SmartInherentProvider as InherentProvider<B>>::get_inherent_providers_and_pre_digest(&SmartInherentProvider {
+                     blocktime: Duration::from_secs(6),
+                 }, maybe_parent_info, parent_header, ext)
+            }
+        }
+    }
+}
+
+/// Attempts to provide inherents in a fashion that works for as many chains as possible.
+///
+/// It is currently tested for
+/// - Polkadot-based relay chains
+/// - Polkadot-ecosystem system parachains
+///
+/// If it does not work for your Substrate-based chain, [please open an issue](https://github.com/paritytech/try-runtime-cli/issues)
+/// and we will look into supporting it.
+struct SmartInherentProvider {
+    blocktime: Duration,
+}
+
+impl<B: BlockT> InherentProvider<B> for SmartInherentProvider {
+    type Err = String;
+
+    fn get_inherent_providers_and_pre_digest(
+        &self,
+        maybe_parent_info: Option<(InherentData, Digest)>,
+        parent_header: B::Header,
+        ext: &mut TestExternalities<HashingFor<B>>,
+    ) -> Result<(Box<dyn sp_inherents::InherentDataProvider>, Vec<DigestItem>), Self::Err> {
+        let blocktime_millis = self.blocktime.as_millis() as u64;
+
+        let timestamp_idp = custom_idps::timestamp::InherentDataProvider {
+            blocktime_millis,
+            maybe_parent_info,
+        };
+        let para_parachain_idp = custom_idps::para_parachain::InherentDataProvider::<B> {
+            blocktime_millis,
+            parent_header: parent_header.clone(),
+            timestamp: timestamp_idp.timestamp(),
+            maybe_para_id: custom_idps::para_parachain::get_para_id::<B>(ext),
+            maybe_last_relay_chain_block_number:
+                custom_idps::para_parachain::get_last_relay_chain_block_number::<B>(ext),
+        };
+        let relay_parachain_data_idp =
+            custom_idps::relay_parachains::InherentDataProvider::<B>::new(parent_header);
+
+        let slot = Slot::from_timestamp(
+            timestamp_idp.timestamp(),
+            SlotDuration::from_millis(blocktime_millis),
+        );
+        let digest = vec![
+            DigestItem::PreRuntime(
+                BABE_ENGINE_ID,
+                PreDigest::SecondaryPlain(SecondaryPlainPreDigest {
+                    slot,
+                    authority_index: 0,
+                })
+                .encode(),
+            ),
+            DigestItem::PreRuntime(AURA_ENGINE_ID, slot.encode()),
```

Comment:
Could you try with some ecosystem runtimes? I assume that moonbeam would not work, but maybe some others could be used to test it.

Reply:
Added `--blocktime` arg to make this work for other runtimes like Aleph Zero that have blocktime other than 6s.

Reply:
Disapointingly, more parachain runtimes I tried than not failed to even compile with the `try-runtime` feature 😅

Out of those that did compile:

Success
- Aleph Zero
- Polimec
- Zeitgeist

Failed
- HydraHX - Digest issue

Unknown
- Acala - couldn't find a node that would serve the date

## Findings

HydraDX issue seems specific to them, I think best to work out later when in contact with their devs who hopefully know what's up with their digest.

Parachains are 12s, relay chain 6s, Aleph Zero 1s. If you get the blocktime wrong, you get a potentially confusing error. So I'm going to remove the default for `--blocktime` so it must be included explicitly.

Reply:
Yea a lot of parachains also dont compile with the benchmark features 🙈 

But at least it shows that it can work. I think that is good enough for now 👍